lib/pymedphys/_experimental/streamlit/apps/mosaiq_claude_chat/app.py

```diff
@@ -49,46 +42,62 @@
 def main():
     _key_handling()
 
-    portal = get_streamlit_trio_portal()
-    messages = _get_messages()
-    message_send_channel = _get_message_send_channel(portal=portal, messages=messages)
+    _initialise_state()
 
     with st.sidebar:
-        _transcript_downloads(messages)
-
-        bak_filepath = (
-            pathlib.Path(
-                st.text_input(".bak file path", value="~/mosaiq-data/db-dump.bak")
-            )
-            .expanduser()
-            .resolve()
+        if st.button("Fill database with CSV records"):
+            create_db_with_tables_from_csv()
+
+        anthropic_api_limit = int(
+            st.number_input("Anthropic API concurrency limit", value=2)
         )
-        if st.button("Start demo MOSAIQ server from .bak file"):
-            start_mssql_docker_image_with_bak_restore(
-                bak_filepath=bak_filepath,
-                sa_password=os.getenv("MSSQL_SA_PASSWORD"),
-            )
 
-    for message in messages:
-        write_message(message["role"], message["content"])
+        if st.button("Remove last message"):
```

Comment:
Does removing a previous message just affect the display to the user and the transcript, or does this "roll back" the conversation (e.g. un-ask a question, or eliminate a previous assertion on the part of the human/user).

Reply:
This quite literally unasks the previous question.  Removing 1 can be used to re-roll the ai response. Removing 2 lets you ask something different.